tests/linearizability/watch.go

```diff
@@ -82,40 +82,73 @@ func validateWatchResponses(t *testing.T, responses [][]watchResponse, expectPro
 }
 
 func validateMemberWatchResponses(t *testing.T, responses []watchResponse, expectProgressNotify bool) {
-	var (
-		gotProgressNotify          = false
-		lastEventModRevision int64 = 1 // The event.Kv.ModRevision in the latest event.
-		lastHeadRevision     int64 = 1 // The resp.Header.Revision in last watch response.
-	)
+	// Validate watch is correctly configured to ensure proper testing
+	validateGotProgressNotify(t, responses, expectProgressNotify)
+
+	// Validate etcd watch properties
+	validateDuplicateHeaderRevisionMeansNoEvents(t, responses)
+	validateResponseHeaderRevisionNeverDecreases(t, responses)
+	validateModRevisionIsStrictlyIncremental(t, responses)
+	validateHeaderRevisionIsLargerThanModRevision(t, responses)
+}
+
+func validateDuplicateHeaderRevisionMeansNoEvents(t *testing.T, responses []watchResponse) {
+	var lastHeadRevision int64 = 1
+	for _, resp := range responses {
+		if resp.Header.Revision == lastHeadRevision && len(resp.Events) != 0 {
+			t.Errorf("Got two non-empty responses about same revision")
```

Comment:
Don't we want to print the problematic revision. 

In theory it might be useful to correlate it later with e.g. watch-log ?

Reply:
Done